### ISSUE TITLE
default storageclass in vcluster synced to host cluster

### ISSUE THREAD

**Issue by moyuduo** [loft-sh/vcluster#669]
### What happened?

hello! It seems that default storageclass will be synced to host cluster unchanged after enbale storageclass sync, this lead to a problem that serveral vcluster's default storageclass present on host cluster as default storageclass and lead host cluster's default storageclass do not worker.

### What did you expect to happen?

maybe we can mark vcluster default storeageclass as not default after synced to host cluster?

### How can we reproduce it (as minimally and precisely as possible)?

1. create serveral vcluster
2. create default storeageclass in each vcluster
3. get storeclass in host cluster

### Host cluster Kubernetes version

<details>

```console
$ kubectl version
# paste output here
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:25:17Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6", GitCommit:"ad3338546da947756e8a88aa6822e9c11e7eac22", GitTreeState:"clean", BuildDate:"2022-04-14T08:43:11Z", GoVersion:"go1.17.9", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Host cluster Kubernetes distribution

<details>

```
# Write here
v1.23.6
```

</details>

### vlcuster version

<details>

```console
$ vcluster --version
# paste output here
vcluster version 0.6.0
```

</details>

### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
# Write here
k3s
```

</details>

### OS and Arch

<details>

```
OS: linux centos7
Arch:amd64
```
</details>

**Comment by matskiv** [comment]
Thank you for reporting this.
Yes, I agree, synced StorageClasses should have the `storageclass.kubernetes.io/is-default-class` annotation removed.

**Comment by mabhi** [comment]
Hi @matskiv , Can you assign this issue to me.

**Comment by matskiv** [comment]
@mabhi Done :) 
Thank you! :raised_hands:

**Comment by mabhi** [comment]
Hi @matskiv, I had put my comments and observations on the slack channel against this issue. As I was unable to get any further response hence am writing here. I checked the code and found the place where the default storage class annotation is being removed explicitly before the sync of the resource to the host cluster. 
The only place where the `http://storageclass.kubernetes.io/is-default-class`  annotation stays is inside the contents of `kubectl.kubernetes.io/last-applied-configuration` tag, which is set by kubectl apply command. It is not recommended to edit this string and hence would like to know your point of view and if further action is expected.
It would help if you can respond either in the slack channel or here for me to proceed further.
Thank you

**Comment by matskiv** [comment]
@mabhi thank you for looking into this. Seem like no further action is needed here. 
I noticed that the issue reporter uses `vcluster version 0.6.0`, and I found this commit where we started excluding the annotation - https://github.com/loft-sh/vcluster/commit/02790b2833f81a946a8a160899cdb632f3840e7c , so that explains why you were not able to reproduce the problem.

@moyuduo The issue is already fixed in a newer version of vcluster (0.7.0+).
